R/zchunk_L226.en_distribution_USA.R

```diff
@@ -63,165 +78,411 @@ module_gcam.usa_L226.en_distribution_USA_DISABLED <- function(command, ...) {
+        datanew
+    } # end global_energy_to_USA_electd
+
+
+    # Process inputs:
+    L226.SubsectorInterp_en %>%
+      mutate(from.year = as.integer(from.year),
+             to.year = as.integer(to.year)) ->
+      L226.SubsectorInterp_en
+
+
+    # Build tables
+
+    # Supplysector information
+
+    # PART 1: FUEL HANDLING AND DELIVERY SECTORS
+
+    # L226.Supplysector_en_USA: Supply sector information for energy handling and delivery sectors
+    # NOTE: Currently using FERC regions as a proxy for regional energy markets
+    A21.sector %>%
+      select(supplysector, output.unit, input.unit, price.unit, logit.exponent, logit.type) %>%
+      filter(supplysector %in%gcamusa.REGIONAL_FUEL_MARKETS) ->
+      A21.tmp
+
+    A26.sector  %>%
+      select(supplysector, output.unit, input.unit, price.unit, logit.exponent, logit.type) %>%
+      filter(supplysector %in%gcamusa.REGIONAL_FUEL_MARKETS) %>%
+      bind_rows(A21.tmp, .) %>%
+      repeat_add_columns(tibble(region = unique(states_subregions$grid_region))) %>%
+      mutate(logit.year.fillout = min(BASE_YEARS)) ->
+      L226.Supplysector_en_USA
+
+
+    # L226.SubsectorShrwtFllt_en_USA: subsector shareweights of energy handling and delivery
+    L226.Supplysector_en_USA %>%
+      mutate(subsector = supplysector,
+             year.fillout = min(BASE_YEARS),
+             share.weight = gcamusa.DEFAULT_SHAREWEIGHT) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["SubsectorShrwtFllt"]])) ->
+      L226.SubsectorShrwtFllt_en_USA
+
+
+    # L226.SubsectorLogit_en_USA
+    # NOTE: There is only one tech per subsector so the logit choice does not matter
+    L226.SubsectorShrwtFllt_en_USA %>%
+      select(one_of(LEVEL2_DATA_NAMES[["Subsector"]])) %>%
+      mutate(logit.year.fillout = min(BASE_YEARS),
+             logit.exponent = gcamusa.DEFAULT_LOGITEXP,
+             logit.type = NA) %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["SubsectorLogit"]], "logit.type"))) ->
+      L226.SubsectorLogit_en_USA
+
+
+    # L226.TechShrwt_en_USA: technology shareweights of energy handling and delivery
+    # NOTE: can't use stub technologies because these would inherit the wrong energy-inputs
+    L226.SubsectorShrwtFllt_en_USA %>%
+      select(one_of(LEVEL2_DATA_NAMES[["Subsector"]])) %>%
+      repeat_add_columns(tibble(year = MODEL_YEARS)) %>%
+      mutate(technology = subsector,
+             share.weight = gcamusa.DEFAULT_SHAREWEIGHT) %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["TechYr"]], "share.weight"))) ->
+      L226.TechShrwt_en_USA
+
+
+    # L226.TechCoef_en_USA: technology coefficients and market names of energy handling and delivery
+    L226.TechShrwt_en_USA %>%
+      select(one_of(LEVEL2_DATA_NAMES[["TechYr"]])) %>%
+      mutate(minicam.energy.input = supplysector,
+             coefficient = gcamusa.DEFAULT_COEFFICIENT,
+             market.name = gcamusa.DEFAULT_MARKET) ->
+      L226.TechCoef_en_USA
+
+
+    # L226.CostAdj_75USDGJ_FERC_F: grid region specific cost adders
```

Comment:
A quick description of how the cost adders are calculated (based on calculation below) would help here.

Reply:
The comments detailing this are already present, but I've slightly updated to make it clearer that that is what they are